### DISCUSSION TITLE
How to make changes in different languages

### DISCUSSION THREAD

**Discussion by GrandM1961** [givanz/Vvveb#20] [General]
Hello I finished the Dutch translation of Vvveb and made a pull request but I would like to implement it straight into the Vvveb. How to do this and what needs to be altered?

It is looking great by the way and I can't wait to be out of the alpha phase.

**Comment by givanz** [comment]
Hi

Thank you for providing the Dutch translation, as the code is not currently pushed on github and the locale/ structure is different I included the dutch translation in the vvveb.zip available on https://www.vvveb.com

The language support is not finished yet and adding new languages from the interface is not currently working, I manually inserted the dutch language record in `language` table and added the translation files to `locale/nl_NL` and to `public/themes/landing/locale/nl_NL`

To generate getext strings for new code you can use translate.sh (the code needs to be edited and language updated to NL) in the root folder then edit and compile the .po files from `locale/nl_NL` (admin) and `public/themes/landing/locale/nl_NL` (frontend/theme)

After compiling the .po files to .mo apache/php/xampp needs to restarted to load the new files.

I will finish the add new language page and release a new update soon.

The html needs some cleanup to remove html entities like `&ensp;` etc from strings and isolate translatable text with some tags like `<span>` to allow proper generation of translatable texts, until this cleanup will be made some texts will be split into multiple strings or have html entities inside making translation harder.

**Comment by GrandM1961** [comment]
I have tried to install with the credentials given bij the installer so admin: super-admin: ***@***.*** ***@***.***> and the given password, and it throws me this error:

Fatal error: Uncaught Exception: Table 'vvveb.admin' doesn't exist UPDATE admin SET `email` = ***@***.***",`user` = "admin",`display_name` = "Super Admin",`password` = "$2y$11$7OMRbgzo37dkS.2rahTrdu5qAxTBq5kWDmbJm3wvjRNbKjswnQ9RG",`status` = 1 WHERE user = "admin" - UPDATE admin SET `email` = :['admin']['email'],`user` = :['admin']['user'],`display_name` = :['admin']['display_name'],`password` = :['admin']['password'],`status` = :['admin']['status'] WHERE user = :user in /Users/hans/Sites/vvveb/system/db/mysqli.php:395 Stack trace: #0 /Users/hans/Sites/vvveb/storage/model/install/adminsql.php(647): Vvveb\System\Db\Mysqli->execute('UPDATE admin \n\t...', Array, Array) #1 /Users/hans/Sites/vvveb/system/user/admin.php(47): Vvveb\Sql\AdminSQL->edit(Array) #2 /Users/hans/Sites/vvveb/install/controller/index.php(177): Vvveb\System\User\Admin::update(Array, Array) #3 [internal function]: Vvveb\Controller\Index->install() #4 /Users/hans/Sites/vvveb/system/core/frontcontroller.php(191): call_user_func(Array) #5 /Users/hans/Sites/vvveb/system/core/frontcontroller.php(256): Vvveb\System\Core\FrontController::call('Vvveb\\Controlle...', 'install', '/Users/hans/Sit...') #6 /Users/hans/Sites/vvveb/system/core/frontcontroller.php(320): Vvveb\System\Core\FrontController::redirect('Index', 'install') #7 /Users/hans/Sites/vvveb/system/core/startup.php(322): Vvveb\System\Core\FrontController::dispatch() #8 /Users/hans/Sites/vvveb/index.php(126): Vvveb\System\Core\start() #9 /Users/hans/Sites/vvveb/install/index.php(51): include('/Users/hans/Sit...') #10 {main} thrown in /Users/hans/Sites/vvveb/system/db/mysqli.php on line 395

And if I use my own credentials it throws me this error:


Fatal error: Uncaught Exception: Table 'vvveb.admin' doesn't exist UPDATE admin SET `email` = ***@***.***",`user` = "admin",`display_name` = "Tormentor",`password` = "$2y$11$IVzLZCBO7gR.XIU17M0dbOxqjR3WDrUEGP2Fb.0cSVvGFx02bRDne",`status` = 1 WHERE user = "admin" - UPDATE admin SET `email` = :['admin']['email'],`user` = :['admin']['user'],`display_name` = :['admin']['display_name'],`password` = :['admin']['password'],`status` = :['admin']['status'] WHERE user = :user in /Users/hans/Sites/vvveb/system/db/mysqli.php:395 Stack trace: #0 /Users/hans/Sites/vvveb/storage/model/install/adminsql.php(647): Vvveb\System\Db\Mysqli->execute('UPDATE admin \n\t...', Array, Array) #1 /Users/hans/Sites/vvveb/system/user/admin.php(47): Vvveb\Sql\AdminSQL->edit(Array) #2 /Users/hans/Sites/vvveb/install/controller/index.php(177): Vvveb\System\User\Admin::update(Array, Array) #3 [internal function]: Vvveb\Controller\Index->install() #4 /Users/hans/Sites/vvveb/system/core/frontcontroller.php(191): call_user_func(Array) #5 /Users/hans/Sites/vvveb/system/core/frontcontroller.php(256): Vvveb\System\Core\FrontController::call('Vvveb\\Controlle...', 'install', '/Users/hans/Sit...') #6 /Users/hans/Sites/vvveb/system/core/frontcontroller.php(320): Vvveb\System\Core\FrontController::redirect('Index', 'install') #7 /Users/hans/Sites/vvveb/system/core/startup.php(322): Vvveb\System\Core\FrontController::dispatch() #8 /Users/hans/Sites/vvveb/index.php(126): Vvveb\System\Core\start() #9 /Users/hans/Sites/vvveb/install/index.php(51): include('/Users/hans/Sit...') #10 {main} thrown in /Users/hans/Sites/vvveb/system/db/mysqli.php on line 395


Also I am running PHP 8.2 and and it says this just before inserting the credentials:

Minimum requirements not met

✖ Recommended PHP version at least 7.0 (You have 5.4) 
✖ Recommended Mysql version at least 7.0 (You have 5.4)


And  also I just changed the username from super-admin to Tormentor and I get to the user login,
No matter if I use my email, old username: super-admin or my newly created username: Tormentoe and the password as the system made it,
Once inserted it says, NOT FOUND: The requested URL was not found on this server and in the browser field it says, localhost/admin//?module=user/login


So I really haven’t a clue what’s going on, I get to the login screen and from there it stops.

**Comment by givanz** [comment]
> Fatal error: Uncaught Exception: Table 'vvveb.admin' doesn't exist 

The admin table error is shown only if `config/db.php` is present and the `admin` table is missing, a very rare case that is checked only to help reinstall on a corrupt database installation.

It's strange that this error is triggered if the `admin` table exists and `config/db.php` is deleted.

>  Recommended PHP version at least 7.0 (You have 5.4)

This is a dummy message from the html template that gets replaced with a real error message if there is an error, if the dummy message is shown it means that there is something wrong with the template display, probably caused by some wrong paths or missing files.

---
Are you trying to install to a subdirectory?  then you need to edit /index.php and set `V_SUBDIR_INSTALL` to the subdirectory name 

```php
defined('V_SUBDIR_INSTALL') || define('V_SUBDIR_INSTALL', '/vvveb');
```

Please redownload to get the latest update and overwrite all files.

The code is tested on php 8.2.1, can you also provide the database version of Mysql or MariaDb?

**Comment by GrandM1961** [comment]
I install it in the root of my server with homebrew and my version of mysql is just updated to 5.7

**Comment by GrandM1961** [comment]
Missing admin table or data, remove config/db.php to reinstall! Table 'vvveb.admin' doesn't exist SELECT * FROM admin AS _ WHERE 1 AND _.role_id = 1 LIMIT 1 - SELECT * FROM admin AS _ WHERE 1 AND _.role_id = :role_id LIMIT 1

Invalid installation. No user with `super admin` role found!

After the first step, I get this but I still can fill in the credentials for website name and the rest for a super-admin account.

Missing admin table or data, remove config/db.php to reinstall! Table 'vvveb.admin' doesn't exist SELECT * FROM admin AS _ WHERE 1 AND _.role_id = 1 LIMIT 1 - SELECT * FROM admin AS _ WHERE 1 AND _.role_id = :role_id LIMIT 1

Invalid installation. No user with `super admin` role found!


I get this after the first attempt:

Fatal error: Uncaught Exception: Table 'vvveb.admin' doesn't exist UPDATE admin SET `email` = ***@***.***",`user` = "admin",`display_name` = "Tormentor",`password` = "$2y$11$4.8n0q02iWgwn.RMRadozOxSadGEoOlf7nHticpVdRQ5xuXrGjQIC",`status` = 1 WHERE user = "admin" - UPDATE admin SET `email` = :['admin']['email'],`user` = :['admin']['user'],`display_name` = :['admin']['display_name'],`password` = :['admin']['password'],`status` = :['admin']['status'] WHERE user = :user in /Users/hans/Sites/vvveb/system/db/mysqli.php:395 Stack trace: #0 /Users/hans/Sites/vvveb/storage/model/install/adminsql.php(647): Vvveb\System\Db\Mysqli->execute('UPDATE admin \n\t...', Array, Array) #1 /Users/hans/Sites/vvveb/system/user/admin.php(47): Vvveb\Sql\AdminSQL->edit(Array) #2 /Users/hans/Sites/vvveb/install/controller/index.php(180): Vvveb\System\User\Admin::update(Array, Array) #3 [internal function]: Vvveb\Controller\Index->install() #4 /Users/hans/Sites/vvveb/system/core/frontcontroller.php(191): call_user_func(Array) #5 /Users/hans/Sites/vvveb/system/core/frontcontroller.php(256): Vvveb\System\Core\FrontController::call('Vvveb\\Controlle...', 'install', '/Users/hans/Sit...') #6 /Users/hans/Sites/vvveb/system/core/frontcontroller.php(320): Vvveb\System\Core\FrontController::redirect('Index', 'install') #7 /Users/hans/Sites/vvveb/system/core/startup.php(322): Vvveb\System\Core\FrontController::dispatch() #8 /Users/hans/Sites/vvveb/index.php(126): Vvveb\System\Core\start() #9 /Users/hans/Sites/vvveb/install/index.php(51): include('/Users/hans/Sit...') #10 {main} thrown in /Users/hans/Sites/vvveb/system/db/mysqli.php on line 395


Db.php is made and all the users are in the base from admin to demo 8 in total and. So I deleted the db.php again and will delete the base also and give it another try.

Second attempt it only shows now a missing image and text saying: These can be later changed from admin.

Attempt to leave everything as suggested except for the email address.

Same result NOT Found
The requested URL was not found on this server.

In the base the email address has changed into mine.

Wil try a last attempt with all the given credentials.
After a third try same again not found the requested URL

Checking dbase now.
I do see something from the frontend of the website but, all a status of 404. And it points to public so I forgot to alter the apache-vvveb.conf to point to my root and vvveb/public.

So one more try.

Same result over and over again.

And when I go to localhost/vvveb/public/index.php a lot of errors saying 404 for images, javascript and a reference error: can’t find variable AOS on index.php:1292


> Op 18 jan. 2023, om 00:28 heeft Givan ***@***.***> het volgende geschreven:
> 
> reinstall

**Comment by givanz** [comment]
Can you please provide more details about the setup you are using?

I retested install on windows 10 xampp and a shared hosting account and it works, I was not able to make the admin error show unless I intentionally delete the table from db.

The 404 errors could be caused by apache not reading the .htaccess or mod_rewrite not enabled.
To enable mod rewrite try `sudo a2enmod rewrite` 

To make sure that .htacess  is loaded and permissions are granted make sure you have in `AllowOverride All` and `Require all granted` in your apache.conf

```xml
<VirtualHost *:80>
	#ServerName www.vvveb.com
	DocumentRoot /var/www/html/vvveb/public

    <Directory />
      Options FollowSymLinks
      AllowOverride All
      Require all granted
    </Directory>

   <Directory /var/www/html/vvveb/public>
        Options Indexes FollowSymLinks
        AllowOverride All
        Require all granted
    </Directory>
	
</VirtualHost>
```
Can you try to install on a shared hosting account or use a package like [xampp](https://www.apachefriends.org/download.html)?

**Comment by GrandM1961** [comment]
I have a mac Studio and i have tried it with MAMP and with homebrew and both give me exactly the same outcome. I will try it later since i deleted homebrew and MAMP and try it again this week.Met vriendelijke groet, HansOp 23 jan. 2023 om 14:19 heeft Givan ***@***.***> het volgende geschreven:﻿
Can you please provide more details about the setup you are using?
I retested install on windows 10 xampp and a shared hosting account and it works, I was not able to make the admin error show unless I intentionally delete the table from db.
The 404 errors could be caused by apache not reading the .htaccess or mod_rewrite not enabled.
To enable mod rewrite try sudo a2enmod rewrite
To make sure that .htacess  is loaded and permissions are granted make sure you have in AllowOverride All and Require all granted in your apache.conf
<VirtualHost *:80>
	#ServerName www.vvveb.com
	DocumentRoot /var/www/html/vvveb/public

    <Directory />
      Options FollowSymLinks
      AllowOverride All
      Require all granted
    </Directory>

   <Directory /var/www/html/vvveb/public>
        Options Indexes FollowSymLinks
        AllowOverride All
        Require all granted
    </Directory>
	
</VirtualHost>
Can you try to install on a shared hosting account or use a package like xampp?

**Comment by GrandM1961** [comment]
I have tried it again but for some reason it points back outside the vvveb folder, it install fine but goes on and of to the root of my localhost and I checked all the configuration files on my webserver and that all points to vvveb/public

**Reply by givanz** [reply]
By pointing back outside you mean it redirects or displays files from the root folder? 
Is it functional and you need to use `/public` for security reasons or there are errors?

Can you please provide more details?
